examples/pytorch/rgcn/entity_sample_cugraph_relgraphconv.py

```diff
@@ -0,0 +1,187 @@
+
+    def update_max_in_degree(self, fanouts):
+        self.conv1.max_in_degree = fanouts[0]
+        self.conv2.max_in_degree = fanouts[1]
+
+
+def evaluate(model, labels, dataloader, inv_target):
+    model.eval()
+    eval_logits = []
+    eval_seeds = []
+    with torch.no_grad():
+        for input_nodes, output_nodes, blocks in dataloader:
+            output_nodes = inv_target[output_nodes.type(torch.int64)]
+            for block in blocks:
+                block.edata["norm"] = dgl.norm_by_dst(block).unsqueeze(1)
+            logits = model(blocks)
+            eval_logits.append(logits.cpu().detach())
+            eval_seeds.append(output_nodes.cpu().detach())
+    eval_logits = torch.cat(eval_logits)
+    eval_seeds = torch.cat(eval_seeds)
+    return accuracy(eval_logits.argmax(dim=1), labels[eval_seeds].cpu()).item()
+
+
+def train(device, g, target_idx, labels, train_mask, model, fanouts):
+    # Define train idx, loss function and optimizer.
+    train_idx = torch.nonzero(train_mask, as_tuple=False).squeeze()
+    loss_fcn = nn.CrossEntropyLoss()
+    optimizer = torch.optim.Adam(model.parameters(), lr=1e-2, weight_decay=5e-4)
+    # Construct sampler and dataloader.
+    sampler = MultiLayerNeighborSampler(fanouts)
+    train_loader = DataLoader(
+        g,
+        target_idx[train_idx].type(g.idtype),
+        sampler,
+        device=device,
+        batch_size=100,
+        shuffle=True,
+    )
+    # No separate validation subset, use train index instead for validation.
+    val_loader = DataLoader(
+        g,
+        target_idx[train_idx].type(g.idtype),
+        sampler,
+        device=device,
+        batch_size=100,
+        shuffle=False,
+    )
+    model.train()
+    for epoch in range(100):
+        total_loss = 0
+        for it, (input_nodes, output_nodes, blocks) in enumerate(train_loader):
+            output_nodes = inv_target[output_nodes.type(torch.int64)]
+            for block in blocks:
+                block.edata["norm"] = dgl.norm_by_dst(block).unsqueeze(1)
+            logits = model(blocks)
+            loss = loss_fcn(logits, labels[output_nodes])
+            optimizer.zero_grad()
+            loss.backward()
+            optimizer.step()
+            total_loss += loss.item()
+        acc = evaluate(model, labels, val_loader, inv_target)
+        print(
+            "Epoch {:05d} | Loss {:.4f} | Val. Accuracy {:.4f} ".format(
+                epoch, total_loss / (it + 1), acc
+            )
+        )
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+        description="RGCN for entity classification with sampling"
+    )
+    parser.add_argument(
+        "--dataset",
+        type=str,
+        default="aifb",
+        help="Dataset name ('aifb', 'mutag', 'bgs', 'am'), default to 'aifb'.",
+    )
+    parser.add_argument(
+        "--idtype",
+        type=str,
+        default="int64",
+        help="Graph idtype ('int64' or 'int32'), default to 'int64'.",
+    )
```

Comment:
Can we remove this argument and stick to one particular option? For users, this seems to be irrelevant for using cuGraph.